src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -125,4 +149,37 @@ private void configureClient(DeviceConfiguration deviceConfiguration) {
         }
         this.iotDataPlaneClient = iotDataPlaneClientBuilder.build();
     }
+
+    /**
+     * Getter for IoT data plane client. This configures the client everytime the getter is used.
+     *
+     * @return iotDataPlaneClient
+     * @throws IoTDataPlaneClientCreationException exception during client configuration
+     */
+    public IotDataPlaneClient getIotDataPlaneClient() throws IoTDataPlaneClientCreationException {
+        configureClient(clientCreationException.get().isPresent());
+        Optional<Exception> exception = clientCreationException.get();
+        if (exception.isPresent()) {
+            throw new IoTDataPlaneClientCreationException(exception.get());
```

Comment:
why are you just throwing the exception?

The key provider may be available by now. This method should try to get a configured client if we don't have one available.

Reply:
Yeah, this is even after trying to configure the client again in L160.

Reply:
OK. Not sure holding the exception is needed. Just using `iotDataPlaneClient == null` or not would seem to accomplish the same in a less complicated way.

Reply:
Removed exception.